### PR TITLE
Expand use of _cleanup_close_ where trivial

### PR DESCRIPTION
**Pull request by vcaputo**
Incentivized by the simplification of take_fdopen/take_fdopendir, convert more call sites to use _cleanup_close_.

### CODE REVIEW

**Comment by vcaputo** [comment]
This builds upon https://github.com/systemd/systemd/pull/15276, if that lands, this adds the more delicate/invasive changes adding _cleanup_close_ where it wasn't previously used.  Review and merge https://github.com/systemd/systemd/pull/15276 before this, or close em all if it's noise.

File: src/basic/fileio.c [hunk 1]
```diff
@@ -54,6 +54,32 @@ int fdopen_unlocked(int fd, const char *options, FILE **ret) {
...
+
+        return 0;
+}
+
+FILE * take_fdopen(int *fd, const char *options) {
+        assert(fd);
+
+        FILE *f = fdopen(*fd, options);
+        if (!f)
+                return NULL;
```
Comment (poettering) [inline src/basic/fileio.c:76]: hmm, if we do a wrapper like this, maybe use our usual way for propagating errors here, too? i.e. like fdopen_unlocked() is not a 1:1 wrapper around fdopen() either, it changes the return handling to follow our own usual style (i.e. kernel-style negative errno returns), so we should do the same for this wrapper too?

**Comment by poettering** [comment]
Looks great, my comment above is just an idea, doesn#t really matter, we currently don't follow what I propose there strictly anyway, hence would merge this either way...

I love the concept!

File: src/basic/fileio.c (hunk 1 above)
Comment (vcaputo) [reply src/basic/fileio.c:76]: Eh, why go mucking with the call sites more than necessary?  It's not like the wrapper is making multiple calls that clobber errno or anything to guard against, so I'm inclined to keep it this way and if we feel like pursuing consistent-ifying everything in the future then we can go rearrange call sites more.

Comment (poettering) [reply src/basic/fileio.c:76]: ok, then let's merge as soon as CI passes. I figure we can just leave this PR open and close the other one?

Comment (vcaputo) [reply src/basic/fileio.c:76]: Sure

File: src/basic/fileio.c [hunk 2]
```diff
@@ -80,6 +80,18 @@ FILE * take_fdopen(int *fd, const char *options) {
         return f;
 }
 
+DIR * take_fdopendir(int *dfd) {
```
Comment (keszybz) [inline src/basic/fileio.c]: `DIR* ` would be nicer.

Comment (vcaputo) [reply src/basic/fileio.c]: Oops, fixed, for take_fdopen() in the .c file too.

**Pull request merged**